### ISSUE TITLE
request to ignore date in other calendar in gramps

### ISSUE THREAD

**Issue by grocanar** [claws/gramps2gource#1]
if one has even with date in other clendar than gregorian the script crash.

i was wondering in the code when we can omit event with date with a cformat attribute

**Comment by claws** [comment]
Can you give me a stack trace of the crash output and perhaps an example of the date format that is causing the problem?

**Comment by grocanar** [comment]
Hi

here is the output of the error
python gramps2gource.py --name="Eric Doutreleau" --db=data.gramps
INFO - Loading Gramps database from data.gramps
INFO - Generating pedigree output for: Eric Doutreleau
ERROR - Problem date '0006-02-30':
Traceback (most recent call last):
   File "gramps2gource.py", line 349, in <module>
     g2g.pedigree(args.names, args.output)
   File "gramps2gource.py", line 136, in pedigree
     associated_events = person.associated_events()
   File "/home/grocanar/gramps2gource/gramps.py", line 323, in 
associated_events
     if event.datetime:
   File "/home/grocanar/gramps2gource/gramps.py", line 136, in datetime
     raise Exception(ex)
Exception: day is out of range for month

and here is the "faulty" lines

<type>Death</type>
       <dateval val="0006-02-30" cformat="French Republican"/>
       <place hlink="_bc0d05de67f10827ae3"/>

as you can see the data is in the French republican format.
the nicest thing would be to convert the date but one can just ignore 
the date in order not to have the exception raised.

**Comment by claws** [comment]
Unfortunately I could not just ignore the date or catch the date conversion exception and return an empty date. Undated events are explicitly filtered out in the `pedigree` function. It does not make sense to allow undated events to propagate through to the output because I don't have any reliable method to place the events into order. 

I have created an update in a branch called `allow_custom_date_handlers`. The change candidate provides the user with the ability to register their own date parser that will be called whenever a matching `cformat` is encountered. If no `cformat` is specified with the gramps date item or there is no date parser registered for the `cformat` then a default date parser will be called. You should be able to create and register you own date parser for the `French Republican` cformat type.

To test if this change works for your use case could you test the candidate change? To do this you will first need to pull the `allow_custom_date_handlers` branch. You can do this using:

```
$ cd <directory to store temporary clone>
$ git clone -b allow_custom_date_handlers https://github.com/claws/gramps2gource.git temp_g2g
$ cd temp_g2g
```

Instead of running `gramps2gource.py` use `custom_date_g2g.py`. You can use the same command line arguments. E.g. `python custom_date_g2g.py --name="Eric Doutreleau" --db=data.gramps`.

However, prior to running the script you will need to implement the `my_date_handler` function to perform the action necessary to convert a `French Republican` date string into a valid datetime. You then need to modify the line where the date parser function is registered with the `gramps` module. So you will probably change the line from:
`gramps.date_processor.register('my_cal_format', my_date_handler)` 
to
`gramps.date_processor.register('French Republican', my_date_handler)`.

Let me know how you go.

**Comment by claws** [comment]
Proposed modification was merged in #2.
